File: sample/src/main/java/com/ms_square/debugoverlay/sample/MainActivity.java
package com.ms_square.debugoverlay.sample;

import static java.lang.Math.max;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.ViewGroup;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import timber.log.Timber;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        EdgeToEdge.enable(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(view -> {
            Timber.d("fab clicked");
            startActivity(new Intent(MainActivity.this, ScrollingActivity.class));
        });

        // Handle FAB insets for navigation bar
        ViewCompat.setOnApplyWindowInsetsListener(fab, (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            ViewGroup.MarginLayoutParams mlp = (ViewGroup.MarginLayoutParams) v.getLayoutParams();
            mlp.leftMargin = max(mlp.leftMargin, systemBars.left);
            mlp.bottomMargin = max(mlp.bottomMargin, systemBars.bottom);
            mlp.rightMargin = max(mlp.rightMargin,systemBars.right);
            v.setLayoutParams(mlp);
            return insets;
        });

        Timber.d("onCreate() called");
    }

    @Override
    public void onResume() {
        super.onResume();
        Timber.d("onResume() called");
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: sample/src/main/kotlin/com/ms/square/debugoverlay/sample/MainActivity.kt
package com.ms.square.debugoverlay.sample

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.fragment.app.FragmentActivity
import com.ms.square.debugoverlay.sample.ui.MainScreen
import com.ms.square.debugoverlay.sample.ui.theme.AppTheme
import dagger.hilt.android.AndroidEntryPoint
import timber.log.Timber

/**
 * Main activity for the DebugOverlay sample app.
 * Hosts the bottom navigation with Feed (Android Weekly RSS) and Overlay Tests tabs.
 */
@AndroidEntryPoint
class MainActivity : FragmentActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    enableEdgeToEdge()

    Timber.tag("MainActivity").d("onCreate() called")

    setContent {
      AppTheme {
        MainScreen()
      }
    }
  }

  override fun onResume() {
    super.onResume()
    Timber.tag("MainActivity").d("onResume() called")
  }
}
